modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,196 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-creating-a-service-from-a-virtual-machine_{context}"]
+
+= Creating a service from a virtual machine
+
+Create a service from a running virtual machine by first creating a Service object
+to expose the service. Connect to the service by specifying the service name.
+
+The service exposes the host name of the virtual machine to the internal cluster,
+using the `ClusterIP` service, or externally, using the `NodePort` or `LoadBalancer`
+service.
+
+This procedure presents an example of how to create, connect to, and expose a Service object
+of `type: ClusterIP` as a virtual machine-backed service.
+
+[NOTE]
+====
+ClusterIP is the default service `type`, if the service `type` is not specified.
+====
+
+.Procedure
+
+. Edit the `VirtualMachine` YAML configuration file.
+. Add the label `special: key` in the `metadata` section.
++
+
+[NOTE]
+====
+Labels on a `VirtualMachine` are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels in
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
++
+
+[source,yaml]
+----
+aapiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachine
+metadata:
+  name: vm-ephemeral
+  namespace: example-namespace
+spec:
+  running: false
+  template:
+    metadata:
+      labels:
+        special: key <1>
+    spec:
+      domain:
+        devices:
+          disks:
+            - name: containerdisk
+              disk:
+                bus: virtio
+            - name: cloudinitdisk
+              disk:
+                bus: virtio
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+        - name: default
+          pod: {}
+      volumes:
+        - name: containerdisk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo
+        - name: cloudinitdisk
+          cloudInitNoCloud:
+            userData: |
+              #!/bin/bash
+              echo "fedora" | passwd fedora --stdin
+----
+<1> The `special: key` label
```

Comment:
Maybe make into a full sentence and add a period?  

```
 <1> The `special: key` label to add.
```

Reply:
@bergerhoffer - Would "Add the `special: key` label" work? "To add" at the end seems to dangle. But if it can't be a directive as a foornote it's OK with me.

Reply:
Nope, your suggestion is just fine!